### PR TITLE
HOSTEDCP-1008: Add NodePoolTransitionSeconds metric

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
records time in seconds it took from NodePool creation until a given condition transitions to true, for the following conditions:
- NodePoolReachedIgnitionEndpoint
- NodePoolAllMachinesReadyConditionType
- NodePoolAllNodesHealthyConditionType

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #https://issues.redhat.com/browse/HOSTEDCP-1008

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: hypershift-operator/controllers/nodepool/metrics/metrics.go [hunk 1]
```diff
@@ -34,6 +35,13 @@ var (
 		Help: "Time in seconds it took from initial NodePool creation and rollout of initial version",
 		Name: NodePoolInitialRolloutDurationMetricName,
 	}, labelNames)
+
+	NodePoolTransitionSecondsMetricName = "hypershift_nodepools_transition_seconds"
+	nodePoolTransitionSeconds           = prometheus.NewHistogramVec(prometheus.HistogramOpts{
+		Name:    NodePoolTransitionSecondsMetricName,
+		Help:    "Time in seconds it took from NodePool creation until a given condition transitions to true",
```
Comment (imain) [inline hypershift-operator/controllers/nodepool/metrics/metrics.go:42]: Do we know the "given condition"?  Could this be explained better?

Comment (muraee) [reply hypershift-operator/controllers/nodepool/metrics/metrics.go:42]: we are tracking multiple conditions with this metric, each condition will have its own time series with the condition name/type set as a label.
the help won't show in Grafana UI as far as I know.

**Review by bryan-cox — APPROVED** [review]
/lgtm

**Pull request merged**